ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+import logging
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.plugin_utils.eos_validate_state_utils.avdtestbase import AvdTestBase
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AvdTestAvtPath(AvdTestBase):
+    """
+    AvdTestAvtPath class for AVT (Adaptive Virtual Topology) tests.
+    Validates the status and type of AVT paths for a specified VRF in WAN scenarios.
+    It constructs a list of static peer addresses for each device by searching through
+    router_path_selection.path_groups.static_peers.
+    """
+
+    anta_module = "anta.tests.avt"
+
+    @cached_property
+    def test_definition(self) -> dict | None:
+        """
+        Generates the ANTA test definition for AVT path tests.
+
+        Returns:
+            dict | None: ANTA test definition or None if the configuration is incomplete.
+        """
+        # Retrieve path groups from the structured configuration
+        path_groups = get(self.structured_config, "router_path_selection.path_groups")
+        if not path_groups:
+            LOGGER.info("Path groups are not configured to collect AVT peer information. %s is skipped.", self.__class__.__name__)
+            return None
+
+        # Retrieve AVT profiles from the structured configuration
+        avt_profiles = get(self.structured_config, "router_adaptive_virtual_topology.vrfs")
+        if not avt_profiles:
+            LOGGER.info("AVT profiles are not configured for any VRF. %s is skipped.", self.__class__.__name__)
+            return None
+
+        # Build a set of static peers
+        static_peers = sorted(
+            {
+                peers_data["router_ip"]
+                for group_idx, path_group in enumerate(path_groups)
+                if self.validate_data(data=path_group, data_path=f"router_path_selection.path_groups.[{group_idx}]", required_keys="static_peers")
```

Comment:
Is it possible to make this block easy using for loop.

Reply:
can we even maybe make it a helper function? like `_get_static_peers` or whaterver?